test/restart/mzcompose.py

```diff
@@ -395,12 +395,13 @@ def workflow_bound_size_mz_status_history(c: Composition) -> None:
     )
 
     # Fill mz_source_status_history and mz_sink_status_history up with enough events
-    for i in range(10):
+    for i in range(5):
         c.testdrive(
             service="testdrive_no_reset",
             input=dedent(
                 """
-                > ALTER CONNECTION kafka_conn SET (SECURITY PROTOCOL PLAINTEXT)
+                > ALTER CONNECTION kafka_conn SET (BROKER 'dne') WITH (VALIDATE = false);
+                > ALTER CONNECTION kafka_conn SET (BROKER '${testdrive.kafka-addr}') WITH (VALIDATE = true);
```

Comment:
Yes––this test requires that we restart the source some number of times to generate events. This PR changes the way altering a connection works––it doesn't actually restart the ingestion if the description matches (because it doesn't meaningfully change anything, so restarting the ingestion is pointless).